src/socket.c

```diff
@@ -54,20 +61,27 @@
 	} \
 } while(0)
 
-#define PTHREADS_SOCKET_ERROR() do { \
-	int eno = php_socket_errno(); \
-	char *estr = eno != SUCCESS ? \
-		php_socket_strerror(eno, NULL, 0) : \
-		NULL; \
-	zend_throw_exception_ex(spl_ce_RuntimeException, eno, \
-		"Error (%d): %s", eno, estr ? estr : "unknown"); \
-	if (eno != SUCCESS) { \
-		if (estr) { \
-			efree(estr); \
+#define PTHREADS_HANDLE_SOCKET_ERROR(eno) do { \
+	if ((eno) != EAGAIN && (eno) != EWOULDBLOCK && (eno) != EINPROGRESS && (eno) != SOCK_EINVAL) { \
+		char *estr = (eno) != SUCCESS ? \
+			php_socket_strerror((eno), NULL, 0) : \
+			NULL; \
+		zend_throw_exception_ex(spl_ce_RuntimeException, (eno), \
+			"Error (%d): %s", (eno), estr ? estr : "unknown"); \
+		if ((eno) != SUCCESS) { \
+			if (estr) { \
+				efree(estr); \
+			} \
 		} \
+		return; \
```

Comment:
I could be wrong, but won't this return statement prevent the `return_value` being set to `false` later on?
https://github.com/krakjoe/pthreads/blob/f15c95e65d52d9f3770094fea199b4ecdb036cae/src/socket.c#L741-L743

Reply:
Also in the `PTHREADS_SOCKET_ERROR` call sites.

Reply:
In case of an exception, there is no need to set a return value or not?

Reply:
Quite true. I overlooked that part...